### DISCUSSION TITLE
JWT Response from auth/local

### DISCUSSION THREAD

**Discussion by Imran109** [strapi/strapi#7504] [Legacy Topics]
Is it possible to only get **JWT** alone or along with **username and id** instead of getting other information like created_by, updated_by and all those as a response? I need to get only the **"user"** field.

**Comment by lauriejim** [comment]
You can deal with that by updating the `callback` function of the `users-permissions` plugin.

Please for that you will have to check how the **extensions** folder work.
**📚Documentation here:** https://strapi.io/documentation/v3.x/concepts/customization.html

And after you will have to find the path of the file you want to update.
It will be this one https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/controllers/Auth.js
So you will have to create an empty file on the same path in the extension folder.
It will be `./extensions/users-permissions/controllers/Auth.js`
```js
module.exports = {
};
```

In this file, you will have to create the function you want to update/override.
It will be this function `callback`
So you have to copy this function and past it in your extensions file.

When it's done you can modify the function as you want.

In your case your will replace:
```js
ctx.send({
  jwt: strapi.plugins['users-permissions'].services.jwt.issue({
    id: user.id,
  }),
  user: sanitizeEntity(user.toJSON ? user.toJSON() : user, {
    model: strapi.query('user', 'users-permissions').model,
  }),
});
```
by 
```js
ctx.send({
  jwt: strapi.plugins['users-permissions'].services.jwt.issue({
    id: user.id,
  })
});
```

**Reply by Imran109** [reply]
There are more than one instances of 

```
ctx.send({
  jwt: strapi.plugins['users-permissions'].services.jwt.issue({
    id: user.id,
  }),
  user: sanitizeEntity(user.toJSON ? user.toJSON() : user, {
    model: strapi.query('user', 'users-permissions').model,
  }),
});
```

Should I replace all ?

**Reply by derrickmehaffy** [reply]
It largely depends on where you only want to get the JWT and not the user data, for example it wouldn't be very useful to cut the user data out of the `/users/me` endpoint which is specifically used to get the current users data, you will want to check what function the other instances are apart of.
